### PR TITLE
feat: add mitex web

### PR DESCRIPTION
**Pull request by Enter-tainer**
image

image

### CODE REVIEW

File: crates/mitex-web-wasm/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "mitex-wasm-web"
```
Comment (Myriad-Dreamin) [inline crates/mitex-web-wasm/Cargo.toml:2]: Can we merge `mitex-web-wasm` and `mitex-typst` into `mitex-wasm`?
I feel okay on build typst plugin with 
```bash
cargo build --bin mitex-wasm --feature typst-plugin
```
And web wasm with

```bash
wasm build --feature web
```

File: crates/mitex-web-wasm/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen::prelude::*;
+
+/// Converts the A LaTex math equation into a plain text
+///
+/// # Errors
+/// Returns an error if the input is not a valid utf-8 string
+/// Returns an error if the input is not a valid math equation
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen)]
+pub fn convert_math(input: &str, spec: &[u8]) -> Result<String, String> {
```
Comment (Myriad-Dreamin) [inline crates/mitex-web-wasm/src/lib.rs:14]: This is a same function as the one in mitex-typst. And we have to maintain comments on both. This is why I want to merge them.

**Review by Myriad-Dreamin — COMMENTED** [review]
Overall looks nice.

File: packages/mitex-web/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    const res = `#math.equation(eval("$" + \`${output.value}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)`;
+    navigator.clipboard.writeText(res);
+  };
+  copy_template_with_imports_button.onclick = () => {
+    update_output();
+    const res = `#import "@preview/mitex:0.1.0": *
+
+#math.equation(eval("$" + \`${output.value}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)`;
+    navigator.clipboard.writeText(res);
+  };
```
Comment (Myriad-Dreamin) [inline packages/mitex-web/src/main.ts:110]: these bindings can be merged into there definitions. For example, 
```ts
  const output = textarea({
    ...
    oninput: update_output,
  });
```

File: packages/mitex-web/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+#math.equation(eval("$" + \`${out}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)
+`;
+
+  /// The source code state
+  const src = van.state(
+    previewTmpl(convert_math(srcDefault, new Uint8Array()))
+  );
+
+  const error = div({ class: "error" });
+  const update_output = () => {
```
Comment (Myriad-Dreamin) [inline packages/mitex-web/src/main.ts:86]: camel case `updateOutput`.

File: scripts/build.sh [hunk 5]
```diff
@@ -1,4 +1,7 @@
 #!/usr/bin/env bash
-cargo build --release --target wasm32-unknown-unknown -p mitex-typst
-rm -f packages/mitex/mitex.wasm
-mv target/wasm32-unknown-unknown/release/mitex_typst.wasm packages/mitex/mitex.wasm
+cargo build --release --target wasm32-unknown-unknown --manifest-path ./crates/mitex-wasm/Cargo.toml --features typst-plugin
+cp target/wasm32-unknown-unknown/release/mitex_wasm.wasm packages/mitex/mitex.wasm
+
+cd crates/mitex-wasm
+wasm-pack build --release --features web
```
Comment (Myriad-Dreamin) [inline scripts/build.sh:6]: Some developers may have not installed wasm-pack. So I think there is a good point to separate a build-web.sh. But I also don't like continuing to increase the number of scripts.

@Enter-tainer and @OrangeX4

**Review by Myriad-Dreamin — COMMENTED** [review]
My final bikeshedding.

File: packages/mitex-web/src/main.ts (hunk 3 above)
Comment (Enter-tainer) [reply packages/mitex-web/src/main.ts:110]: update_output uses input_area and previewTmpl, which has not been defined when defining buttons and areas.

Comment (Enter-tainer) [reply packages/mitex-web/src/main.ts]: the on focus handler is moved

File: scripts/build.sh (hunk 5 above)
Comment (Enter-tainer) [reply scripts/build.sh:7]: > Some developers may have not installed wasm-pack

maybe just install them.

File: packages/mitex-web/src/main.ts (hunk 3 above)
Comment (Myriad-Dreamin) [reply packages/mitex-web/src/main.ts]: What about using some state bindings and placing them ahead?

**Review by Myriad-Dreamin — APPROVED** [review]

**Pull request merged**